src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -188,7 +188,7 @@ export async function generateViewFunctionPayload(args: InputViewFunctionDataWit
   });
 
   // Fill in the ABI
-  return generateViewFunctionPayloadWithABI({ abi: functionAbi, ...args });
+  return generateViewFunctionPayloadWithABI({ ...args, abi: functionAbi });
```

Comment:
why are we doing it that way if view functions can take in json arguments? when there is no abi but json arguments we should not do any server call

Reply:
View functions take in BCS arguments?

The JSON parser on the server side isn't actually well featured enough to support most inputs.

If we really want to support the JSON inputs, we can, but it's the same problem as why transactions are encoded in BCS rather than JSON.

Reply:
https://aptos-org.slack.com/archives/C05NLAKJM9U/p1718147823459949?thread_ts=1718144833.994919&channel=C05NLAKJM9U&message_ts=1718147823.459949